osu.Game/Screens/Select/Details/AdvancedStats.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using System;
+using osu.Game.Beatmaps;
+
+namespace osu.Game.Screens.Select.Details
+{
+    public class AdvancedStats : Container
+    {
+        private readonly StatisticRow firstValue, hpDrain, accuracy, approachRate, starDifficulty;
+
+        private BeatmapInfo beatmap;
+        public BeatmapInfo Beatmap
+        {
+            get { return beatmap; }
+            set
+            {
+                if (value == beatmap) return;
+                beatmap = value;
+
+                //mania specific
+                if ((Beatmap?.Ruleset?.ID ?? 0) == 3)

```

Comment:
This is fine for now, but going forward we should be exposing these (and all other difficulty information below) via the `Ruleset` interface, in a similar way to `GetBeatmapStatistics`